cpp/src/io/orc/writer_impl.cu

```diff
@@ -1803,7 +1808,7 @@ void writer::impl::write(table_view const& table)
   auto dictionaries = allocate_dictionaries(orc_table, rowgroup_bounds, stream);
   hostdevice_2dvector<gpu::DictionaryChunk> dict(
     rowgroup_bounds.size().first, orc_table.num_string_columns(), stream);
-  if (orc_table.num_string_columns() != 0) {
+  if (not dict.is_empty()) {
```

Comment:
Just for my understanding, does this change actually change the behavior? The second argument to the `dict` constructor above is based on the number of string columns, so I'm assuming that checking `dict.is_empty()` is equivalent to looking at the number of string columns in the table?

Reply:
It's not the same when there are string columns, but they have no rows. In this case `dict `is empty because `rowgroup_bounds.size().first` is zero, whereas `orc_table.num_string_columns()` is not.